### PR TITLE
[Frontend] Merge the revised homepage & QoL changes

### LINKED ISSUE #487: [Frontend] Update the homepage to work with new endpoints
After the decision to handle images as base64 from database, the endpoints and handling of image in both backend and frontend instances changed. So in our current implementation, homepage takes image url instead of the returned image id, in which the image component will use to fetch the image from the API. While updating the homepage, I will also update its styling and support other newly added features.

**Deadline: 2022-12-04**
**Reviewer: @guneyizol**

### PR DESCRIPTION
**Pull request by erimerkin**
In this PR the requested changes for issue #487 is implemented. The last version of the page can be seen in https://github.com/bounswe/bounswe2022group7/issues/487#issuecomment-1336383343. Succesful merge of this PR closes #487.

Summary of the changes:
- Renaming the docker-compose containers to `ideart`
- API connection and image handling fix for `/profile/settings` `EditUserInfo` components
- API connection and post handling revision along with style changes for `HomePage`
- Removal of `ArtItem`, `Event`, `ForumPost` components from homepage with a new standard `FeedCard` component taking their places.
- Created tests for `LoadingButton` and `FeedCard` components
- Implementation of toggleable `Chips` for filtering, they can also be used for category/label selection in the future.
- Implementation of standard avatar handler `UserAvatar`
- Implementation of customizable dropdown menu component: `CustomizableDropdownMenu` 
- Cleaning up accumulated errors on components
- Reactivating the CI's testing step
- Setting the color pallete of the application

### CODE REVIEW

**Comment by CahidArda** [comment]
Is it possible to use "Art Item" instead of "Art Work"?

File: frontend/src/pages/HomePage/HomePage.js [hunk 1]
```diff
@@ -1,91 +1,209 @@
...
+    // useEffect(() => {
+    //     const sortedContent = content.sort((a, b) => (a.content.creationDate > b.content.creationDate) ? -1 : 1);
+    //     setContent(sortedContent);
+    // }, [isLoaded])
 
     if (error) {
         return <div>Error: {error.message}</div>
     } else if (!isLoaded) {
-        return <div>Loading...</div>
+        return <div><CircularProgress /></div>
```
Comment (CahidArda) [inline frontend/src/pages/HomePage/HomePage.js:183]: Should I add this to the other pages if it looks better?

Comment (erimerkin) [reply frontend/src/pages/HomePage/HomePage.js:182]: I don't think that it is necessary for now

File: frontend/src/components/UserAvatar.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+    React.useEffect(() => {
+        fetch('/api/image/' + id, { method: "GET" })
+            .then((response) => response.json())
+            .then((json) => setImage(json.base64String))
+            .catch((error) => console.log(error))
+    }, [id])
+
+    return (
+        <Avatar data-testid="userAvatar" alt={username} src={image} sx={{...sx}} />
```
[the remark this answers is not in the document]
Comment (erimerkin) [reply frontend/src/components/UserAvatar.js:17]: It is used in unit testing, with the testid I can easily check if the component is rendered and select it.

File: frontend/src/layouts/GenericCardLayout.js [hunk 3]
```diff
@@ -11,7 +11,7 @@ const styles = {
         display: 'flex',
         flexDirection: 'column',
         alignItems: 'center',
-        marginTop: 4,
+        // marginTop: 4,
```
[the remark this answers is not in the document]
Comment (erimerkin) [reply frontend/src/layouts/GenericCardLayout.js:14]: Removed

File: frontend/src/common/UserCard.js [hunk 4]
```diff
@@ -41,7 +42,8 @@ function UserCard(props) {
   return (
     <Grid container wrap="nowrap" spacing={2}>
       <Grid item>
-        <Avatar alt="User Profile" src={authorAccountInfo.profilePictureId} />
+        {/* <Avatar alt="User Profile" src={authorAccountInfo.profilePictureId} /> */}
```
[the remark this answers is not in the document]
Comment (erimerkin) [reply frontend/src/common/UserCard.js:45]: Removed.

**Comment by erimerkin** [comment]
> Is it possible to use "Art Item" instead of "Art Work"?

I have changed it to Art Item in recent commit aaed8ad

**Review by guneyizol — APPROVED** [review]

**Pull request merged**